Gordon360/Controllers/MembershipsController.cs

```diff
@@ -83,6 +83,21 @@ public ActionResult<int> GetMembershipCount(string? activityCode = null, string?
             return Ok(result);
         }
 
+        /// <summary>
+        /// Gets the memberships of the given user
+        /// </summary>
+        /// <param name="username"> username filter </param>
+        /// <returns>The number of followers of the activity</returns>
+        [HttpGet]
+        [Route("student/{username}")]
+        [StateYourBusiness(operation = Operation.READ_ONE, resource = Resource.MEMBERSHIP)] // not sure why
+        public ActionResult<int> GetMembershipsOfUser(string username)
+        {
+            var result = _membershipService.GetMemberships(username: username, sessionCode: "*");
+
+            return Ok(result);
+        }
+
```

Comment:
I don't understand why this route was added. All it does is implement d subset of the base `/memberships` route, in basically the same way as the `/profiles/{username}/memberships` route. The latter is better in terms of REST API design, but it was marked obsolete because the base `/memberships` route already covers all it's possible use cases with optional query params.

It looks like this was added for https://github.com/gordon-cs/gordon-360-ui/pull/1083, but the better route to use for that should be `/profiles/{username}/memberships-history`, the same way it's currently used for the Experience Transcript.

Reply:
Antonia and Amos confirmed that the new endpoint is not needed, and in the process found an apparent security flaw in the recommended endpoint: if given no username, it returns all involvements for all users, without filtering to only public involvements.  Antonia is working on a fix for both issues.

Reply:
> Antonia and Amos confirmed that the new endpoint is not needed, and in the process found an apparent security flaw in the recommended endpoint: if given no username, it returns all involvements for all users, without filtering to only public involvements. Antonia is working on a fix for both issues.

If you're talking about the GetMembershipHistory method that I recommended (route `/profiles/{username}/memberships-history`), I'm fairly certain it can't be called without the `username` param. If you try, the API sends back this error:
```json
{
  "errors": {
    "username": [
      "The username field is required."
    ]
  },
  "type": "https://tools.ietf.org/html/rfc7231#section-6.5.1",
  "title": "One or more validation errors occurred.",
  "status": 400,
  "traceId": "00-b582e79374a558d2ff3c822dc9ba1d2a-ddb6eacb502fbb4c-00"
}
```

Reply:
Sorry, it's the base /memberships route which appears to have a security flaw.